Native tests fail

https://github.com/micronaut-projects/micronaut-starter/actions/runs/5892460932

Error details

```
--
  | Failures (1): |  
  | JUnit Jupiter:StarterWebNettyTest:testGitHubCreate() |  
  | MethodSource [className = 'example.StarterWebNettyTest', methodName = 'testGitHubCreate', methodParameterTypes = ''] |  
  | => java.io.UncheckedIOException: java.nio.file.NoSuchFileException: /tmp/test-github-create10280145358235947449/gc.log.lock |  
  | java.base@17.0.8/java.nio.file.FileTreeIterator.fetchNextIfNeeded(FileTreeIterator.java:87) |  
  | java.base@17.0.8/java.nio.file.FileTreeIterator.hasNext(FileTreeIterator.java:103) |  
  | java.base@17.0.8/java.util.Iterator.forEachRemaining(Iterator.java:132) |  
  | java.base@17.0.8/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845) |  
  | java.base@17.0.8/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) |  
  | [...] |  
  | Caused by: java.nio.file.NoSuchFileException: /tmp/test-github-create10280145358235947449/gc.log.lock |  
  | java.base@17.0.8/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55) |  
  | java.base@17.0.8/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:148) |  
  | java.base@17.0.8/sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99) |  
  | java.base@17.0.8/java.nio.file.Files.readAttributes(Files.java:1851) |  
  | java.base@17.0.8/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:220) |  
  | [...]
```

--- a/test-suite-graal/src/test/java/example/GitHubApiClientMock.java
+++ b/test-suite-graal/src/test/java/example/GitHubApiClientMock.java
@@ -8,24 +8,63 @@ import io.micronaut.starter.client.github.v3.GitHubSecretsPublicKey;
 import io.micronaut.starter.client.github.v3.GitHubUser;
 import io.micronaut.starter.client.github.v3.GitHubWorkflowRun;
 import io.micronaut.starter.client.github.v3.GitHubWorkflowRuns;
+import jakarta.annotation.PostConstruct;
+import jakarta.annotation.PreDestroy;
 import jakarta.inject.Singleton;
 import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.lib.Repository;
 import org.eclipse.jgit.transport.URIish;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.io.File;
+import java.net.URI;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.Comparator;
 
 @Replaces(GitHubApiClient.class)
 @Singleton
 public class GitHubApiClientMock implements GitHubApiClient {
 
+    private static final Logger LOG = LoggerFactory.getLogger(GitHubApiClientMock.class);
+
+    private Repository gitRepository;
+
+    private String gitRepositoryUrl;
+
+    @PostConstruct
+    public void initialize() throws Exception {
+        Path temp = Files.createTempDirectory("test-github-create");
+        Git git = Git.init().setDirectory(temp.toFile()).setBare(true).call();
+        gitRepository = git.getRepository();
+        URIish uri = new URIish(gitRepository.getDirectory().toURI().toURL());
+        gitRepositoryUrl = uri.toString();
+    }
+
+    @PreDestroy
+    public void close() {
+        if (gitRepository != null) {
+            try {
+                gitRepository.close();
+            } catch (Exception e) {
+                LOG.warn("Failed to close git repository: {}", gitRepositoryUrl, e);
+            }
+            try {
+                Files.walk(Path.of(new URI(gitRepositoryUrl)))
+                        .sorted(Comparator.reverseOrder())
+                        .map(Path::toFile)
+                        .forEach(File::delete);
+            } catch (Exception e) {
+                LOG.warn("Failed to delete git repository: {}", gitRepositoryUrl, e);
+            }
+        }
+    }
+
     @Override
     public GitHubRepository createRepository(String oauthToken, GitHubRepository gitHubRepository) {
         try {
-            Path temp = Files.createTempDirectory("test-github-create");
-            Git repo = Git.init().setDirectory(temp.toFile()).setBare(true).call();
-            URIish uri = new URIish(repo.getRepository().getDirectory().toURI().toURL());
-            return new GitHubRepository(gitHubRepository.getName(), gitHubRepository.getDescription(), "url", "html_url", uri.toString());
+            return new GitHubRepository(gitHubRepository.getName(), gitHubRepository.getDescription(), "url", "html_url", gitRepositoryUrl);
         } catch (Exception e) {
             throw new RuntimeException("Failed to create repository: " + gitHubRepository.getName(), e);
         }

--- a/test-suite-graal/src/test/java/example/StarterWebNettyTest.java
+++ b/test-suite-graal/src/test/java/example/StarterWebNettyTest.java
@@ -12,12 +12,6 @@ import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
 import jakarta.inject.Inject;
 import org.junit.jupiter.api.Test;
 
-import java.io.File;
-import java.net.URI;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.Comparator;
-
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertNull;
@@ -50,15 +44,11 @@ public class StarterWebNettyTest {
     }
 
     @Test
-    void testGitHubCreate() throws Exception {
+    void testGitHubCreate() {
         starterConfiguration.setRedirectUri(null);
         GitHubCreateDTO response = client.toBlocking().retrieve(HttpRequest.GET(GITHUB_CREATE_URI), GitHubCreateDTO.class);
         assertNotNull(response);
         assertNotNull(response.getCloneUrl());
-        Files.walk(Path.of(new URI(response.getCloneUrl())))
-                .sorted(Comparator.reverseOrder())
-                .map(Path::toFile)
-                .forEach(File::delete);
     }
 
     @Test

